src/rabbit_variable_queue.erl

```diff
@@ -1271,48 +1264,102 @@ msg_size(#msg_status{msg_props = #message_properties{size = Size}}) -> Size.
 
 msg_in_ram(#msg_status{msg = Msg}) -> Msg =/= undefined.
 
-remove(AckRequired, MsgStatus = #msg_status {
```

Comment:
What I did here was to try to simplify remove/3 by having two function bodies, one for when `AckRequired=true` the other one for when it's `false`.